sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -217,11 +241,12 @@ def handle_result(
             self._side_inputs_container.add_values(
                 view,
                 committed_bundle.get_elements_iterable(make_copy=True))
-          if (self.get_execution_context(result.transform)
-              .watermarks.input_watermark
-              == WatermarkManager.WATERMARK_POS_INF):
-            self._pending_unblocked_tasks.extend(
-                self._side_inputs_container.finalize_value_and_get_tasks(view))
+
+      tasks = self._watermark_manager.update_watermarks(
```

Comment:
<!--new_thread; commit:2053e91ce5cf41b329a4bf8bdb48a06b79b3626b; resolved:0-->
Please add comment that these tasks come from unblocked side inputs.

Reply:
Done